lib/store/watches.js

```diff
@@ -12,20 +14,23 @@ export default class WatchesStore {
   kernel: Kernel;
   @observable
   watches: Array<WatchStore> = [];
+  autocompleteDisposables: ?atom$CompositeDisposable;
 
   constructor(kernel: Kernel) {
     this.kernel = kernel;
 
     this.kernel.addWatchCallback(this.run);
+    AutocompleteConsumer.register(this.autocompleteDisposables);
```

Comment:
This part causes obvious and serious errors.

AFAIU `this.autocompleteDisposables` in watches.js does nothing other than that currently.
Can you explain your purpose of this `register` stuff ?

Reply:
Oh I remember why the new ones where in disable now too. The purpose of register is to notify the consumer class that this disposable must be disposed when autocomplete is disabled.

Reply:
The error is because now it is no longer predefined I need to create a new disposable right above that line

Reply:
`this.autocompleteDisposables = new CompositeDisposable`

Reply:
I would much rather revert my disable commit though because that way there is always a disposable even if it is empty because there is no autocomplete it will still be there

Reply:
I see.
Don't you think `this.autocompleteDisposables = new CompositeDisposable();` in class declaration is not enough tho ?

Reply:
I am not to sure. I removed it during my disable commit but maybe I shouldn't have. It was originally where the type declaration is.